models/promise.js

```diff
@@ -48,7 +48,7 @@ export { Promise, sequelize }
 // DATABASE FIELDS:
 //   urtx -- urtext, ie, full original text the user typed to create the promise
 //   user -- who's making the promise
-//   what -- what's being promised
+//   slug -- https://en.wikipedia.org/wiki/Semantic_URL#Slug (previously "what")
```

Comment:
@dreeves 👍  for the slug concept although i think the `slug` would actually be the full original text? the idea is that it's like a unique id/key, right? _ah, i see you addressed this in https://github.com/beeminder/iwill/issues/23 so we can continue this there_

sidenote: while i appreciate the cleverness in all the column name lengths <= 4 characters, i'd prefer to have more verbose names in general. so many times i typed `urtext` or `urtxt` when i meant `urtx` but what i really wanted was to say `userText` and be done with it (but maybe i'm just slow and should have thought to check myself against the 4 character rule)

obviously sticking to one convention is more important than picking the 'right' one, but if/as we're reevaluating the data model i think this is worth considering

Reply:
Slugs: It's actually standard terminology that in a URL like "subdomain.domain.tld/foo_bar" that the "foo_bar" is known as the slug. I think it's the right term for us to emphasize that it's the unique identifier for a promise. We used to use the same term in Beeminder before we switched to calling them goalnames.

Urtext: This is also a standard term for "original text" (the "ur" is German for "original") and I really like it. <google.com/search?q=urtext>

Overly short names: I have way too much fun with this kind of thing. I treat it as a fun challenge to make a set of related names all follow the same syntactic constraint. But I confess that abbreviating "urtext" to "urtx" is a bit much. Maybe that one can be the exception...

Reply:
aha, i learned something new today with urtext. and i get it, nomenclature is like our bread and butter or whatever haha

re slugs: i was thinking about them in a site-wide context (like, unique identifier resolving at the root domain) but i get what you're doing

promises will be keyed/IDed on user + (some kind of normalized) `slug`, so of course the 'slug' for a user's promise should be just the parsed promise text

Reply:
Idea for keeping complications at bay: instead of doing any normalizing of slugs, we just reject any URL that doesn't strictly conform to, say, `/^[\w\-]+$/`

The first time we notice that being too restrictive we can revisit it. But it might actually help users, to have a nice simple memorable rule for what's allowed in the URLs.

Reply:
Just to make sure we're on the same page:

If you click on `http://alice.commits.to/foo_bar/by/tomorrow` then this is the promise that's created:

* `urtext`: `alice.commits.to/do_the_thing/by/tomorrow`
* `user`: `alice`
* `slug`: `do_the_thing`

And any new URL with subdomain `alice` and slug `do_the_thing` will refer to that promise.

Reply:
@dreeves looks good to me!